libs/langchain/langchain/memory/chat_message_histories/elasticsearch.py

```diff
@@ -0,0 +1,114 @@
+import json
+import logging
+from time import time
+from typing import TYPE_CHECKING, List
+
+from langchain.schema import BaseChatMessageHistory
+from langchain.schema.messages import BaseMessage, _message_to_dict, messages_from_dict
+
+if TYPE_CHECKING:
+    from elasticsearch import Elasticsearch
+
+logger = logging.getLogger(__name__)
+
+
+class ElasticsearchChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history that stores history in Elasticsearch.
+
+    Args:
+        client: Elasticsearch client.
+        index: name of the index to use.
+        session_id: arbitrary key that is used to store the messages
+            of a single chat session.
+    """
+
+    def __init__(
+        self,
+        client: Elasticsearch,
```

Comment:
i suggest keeping the same API as the ElasticsearchStore for connection. Using this function, you should be able to support a variety of connection configurations for local and cloud

https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/vectorstores/elasticsearch.py#L513

Reply:
Also keep in mind we should customise the user-agent to "langchain-py-ms/{version}" to provide us telemetry.

Reply:
Updated `ElasticsearchChatMessageHistory` to use the same API and added user agent header.